aptos-move/block-executor/src/executor.rs

```diff
@@ -535,6 +533,25 @@ where
                 patched_events.push(event);
             }
         }
+        patched_events
+    }
+
+    fn materialize_txn_commit(
+        &self,
+        txn_idx: TxnIndex,
+        versioned_cache: &MVHashMap<T::Key, T::Tag, T::Value, X, T::Identifier>,
+        scheduler: &Scheduler,
+        shared_counter: &AtomicU32,
+        last_input_output: &TxnLastInputOutput<T, E::Output, E::Error>,
+        base_view: &S,
+        final_results: &ExplicitSyncWrapper<Vec<E::Output>>,
+    ) {
+        let parallel_state = ParallelState::<T, X>::new(versioned_cache, scheduler, shared_counter);
+        let latest_view = LatestView::new(base_view, ViewState::Sync(parallel_state), txn_idx);
+        let patched_resource_write_set =
+            Self::map_id_to_values_in_read_write_set(txn_idx, last_input_output, &latest_view);
+        let patched_events =
+            Self::map_id_to_values_events(txn_idx, last_input_output, &latest_view);
 
         // Materialize all the aggregator v1 deltas.
         let aggregator_v1_delta_keys = last_input_output.aggregator_v1_delta_keys(txn_idx);
```

Comment:
I know it's unrelated to your change, but can you extract agg_v1 handling into their own function as well?